### ISSUE TITLE
False positive when protocol method implementation takes both optional arguments and *args or **kwargs

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1728]
**Describe the bug**
A class implementing a protocol is reported to be incompatible when a method implementation accepts both optional arguments and *args or **kwargs, where the protocol method takes no arguments.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That no error would be produced.

**Screenshots or Code**
```python
from typing import Protocol

class Foo(Protocol):
    def method(self) -> None: ...

class Bar:
    def method(
        self,
        value: str = ...,
    ) -> None: ...

class Baz:
    def method(
        self,
        *args: object,
    ) -> None: ...

class Qux:
    def method(
        self,
        value: str = ...,
        *args: object,
    ) -> None: ...

class Quux:
    def method(
        self,
        value: str = ...,
        **kwargs: object,
    ) -> None: ...

class Quuux:
    def method(
        self,
        value: str = ...,
        *args: object,
        **kwargs: object,
    ) -> None: ...


bar: Foo = Bar()
baz: Foo = Baz()
qux: Foo = Qux()  # Expression of type "Qux" cannot be assigned to declared type "Foo" [...]
quux: Foo = Quux()  # Expression of type "Quux" cannot be assigned to declared type "Foo" [...]
quuux: Foo = Quuux()  # Expression of type "Quuux" cannot be assigned to declared type "Foo" [...]

```

**VS Code extension or command-line**
1.1.128

**Comment by erictraut** [comment]
Thanks for the bug report and the excellent repro steps.

This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.129, which I just published. It will also be included in the next release of pylance.
